aip/cloud/25146.md

```diff
@@ -0,0 +1,25 @@
+---
+id: 25146
+state: draft
+created: TBD
+placement:
+  category: gcp
+---
+
+# Generic Fields
+
+Guidelines related to generic fields in [AIP-146](../0146.md) **should** be applied
+with the following exceptions.
+
+## Maps
+
+Fields of type `map` **must not** be composed of type Message. `map<primitive, primitive>`
```

Comment:
Reading over google.aip.dev/146#maps again, this new AIP seems to be in direct contention with the following example given in the Core AIP:

```
In particular, a map can sometimes be suited to a situation where many objects of the same type are needed, with different behavior based on the names of their keys (for example, using keys as environment names).
```

I think that is still a valid use case. How do we accommodate that?